### PR TITLE
libxx: fix/suppresss warnings

### PR DESCRIPTION
**Pull request by yamt**
## Summary
libxx: fix/suppresss warnings

### CODE REVIEW

**Comment by v01d** [comment]
Maybe since libxx supports exceptions we can implement a trivial bad_alloc exception when exceptions are on

**Comment by yamt** [comment]
@v01d maybe. if CONFIG_CXX_EXCEPTION=y. anyway i'm not going to implement it in this PR.

**Comment by v01d** [comment]
Sure, just mentioning it.

**Review by v01d — APPROVED** [review]

File: libs/libxx/cxx.defs [hunk 1]
```diff
@@ -22,3 +22,14 @@ CXXSRCS += libxx_cxa_guard.cxx libxx_cxapurevirtual.cxx
 CXXSRCS += libxx_delete.cxx libxx_delete_sized.cxx libxx_deletea.cxx
 CXXSRCS += libxx_deletea_sized.cxx libxx_new.cxx libxx_newa.cxx
 CXXSRCS += libxx_stdthrow.cxx
+
+# Note: Our implmenetations of operator new are not conforming to
+# the standard. (no bad_alloc implementation)
+#
+# libxx_new.cxx:64:11: error: 'operator new' is missing exception specification
+#       'throw(std::bad_alloc)' [-Werror,-Wmissing-exception-spec]
```
Comment (xiaoxiang781216) [inline libs/libxx/cxx.defs:30]: But should we implement bad_alloc instead?

Comment (v01d) [reply libs/libxx/cxx.defs:30]: @yamt can you open an issue to remind ourselves to fix this? shouldn't be hard

**Pull request merged**